content/Development/qtest.md

```diff
@@ -0,0 +1,224 @@
+
+| Driver | Query File's Location | Property |
+|-|-|-|
+| TestCliDriver | ql/src/test/queries/clientpositive | mr.query.files |
+| TestMinimrCliDriver | ql/src/test/queries/clientpositive | minimr.query.files |
+| TestMiniTezCliDriver | ql/src/test/queries/clientpositive | minitez.query.files, minitez.query.files.shared |
+| TestMiniLlapCliDriver | ql/src/test/queries/clientpositive | minillap.query.files |
+| TestMiniDruidCliDriver | ql/src/test/queries/clientpositive | druid.query.files |
+| TestMiniDruidKafkaCliDriver | ql/src/test/queries/clientpositive | druid.kafka.query.files |
+| TestMiniHiveKafkaCliDriver | ql/src/test/queries/clientpositive | hive.kafka.query.files |
+| TestMiniLlapLocalCompactorCliDriver | ql/src/test/queries/clientpositive | compaction.query.files |
+| TestEncryptedHDFSCliDriver | ql/src/test/queries/clientpositive | encrypted.query.files |
+| TestBeeLineDriver | ql/src/test/queries/clientpositive | beeline.positive.include, beeline.query.files.shared |
+| TestErasureCodingHDFSCliDriver | ql/src/test/queries/clientpositive | erasurecoding.only.query.files |
+| MiniDruidLlapLocalCliDriver | ql/src/test/queries/clientpositive | druid.llap.local.query.files |
+| TestNegativeLlapCliDriver | ql/src/test/queries/clientnegative | llap.query.negative.files |
+| TestIcebergLlapLocalCliDriver | iceberg/iceberg-handler/src/test/queries/positive | iceberg.llap.query.files |
+| IcebergLlapLocalCompactorCliConfig | iceberg/iceberg-handler/src/test/queries/positive | iceberg.llap.query.compactor.files |
+
+## Advanced
+
+### Locations of log files
+
+The Query Unit Test framework outputs log files in the following paths.
+
+- `itests/{qtest, qtest-accumulo, qtest-iceberg, qtest-kudu}/target/surefire-reports`
+- From the root of the source tree: `find . -name hive.log`
+
+
+### How do I run with Postgre/MySQL/Oracle?
+
+To run a test with a specified DB, it is possible by adding the "-Dtest.metastore.db" parameter like in the following commands:
+
+```sh
+mvn test -Pitests -pl itests/qtest -Dtest=TestCliDriver -Dqfile=partition_params_postgres.q -Dtest.metastore.db=postgres
+mvn test -Pitests -pl itests/qtest -Dtest=TestCliDriver -Dqfile=partition_params_postgres.q -Dtest.metastore.db=mssql
+mvn test -Pitests -pl itests/qtest -Dtest=TestCliDriver -Dqfile=partition_params_postgres.q -Dtest.metastore.db=mysql
+mvn test -Pitests -pl itests/qtest -Dtest=TestCliDriver -Dqfile=partition_params_postgres.q -Dtest.metastore.db=oracle -Ditest.jdbc.jars=/path/to/your/god/damn/oracle/jdbc/driver/ojdbc6.jar
+```
+
+### Remote debug
+
+Remote debugging with Query Unit Test is a potent tool for debugging Hive.
+
+```sh
+$ mvn -Dmaven.surefire.debug="-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=localhost:5005" test -Dtest=TestCliDriver -Dqfile=<test>.q
```

Comment:
In most cases it suffices to just put `-Dmaven.surefire.debug` without a specific value. The default configuration works fine most of the time.

Reply:
+1 for `-Dmaven.surefire.debug`